packages/syft/src/syft/service/worker/worker_service.py

```diff
@@ -182,37 +184,46 @@ def status(
         context: AuthedServiceContext,
         uid: UID,
     ) -> Union[Tuple[WorkerStatus, WorkerHealth], SyftError]:
+        worker = self.get(context=context, uid=uid)
+
+        if not isinstance(worker, SyftWorker):
+            return worker
+
+        return worker.status, worker.healthcheck
+
+    @service_method(
+        path="worker.get",
+        name="get",
+        roles=DATA_SCIENTIST_ROLE_LEVEL,
+    )
+    def get(
+        self, context: AuthedServiceContext, uid: UID
+    ) -> Union[SyftWorker, SyftError]:
         result = self.stash.get_by_uid(credentials=context.credentials, uid=uid)
         if result.is_err():
             return SyftError(message=f"Failed to retrieve worker with UID {uid}")
-        worker: SyftWorker = result.ok()
 
-        if context.node.in_memory_workers:
-            return worker.status, worker.healthcheck
+        worker = cast(SyftWorker, result.ok())
```

Comment:
Can we handle another condition here, if `result.ok()` is None then return `SyftError(message=f"Worker doesn't exists for id: {uid}")`

Reply:
Resolve by 01c89f3